### ISSUE TITLE
[FR] UI: Download artifacts without loading them in the browser

### ISSUE THREAD

**Issue by ntakouris** [mlflow/mlflow#7508]
### Willingness to contribute

No. I cannot contribute this feature at this time.

### Proposal Summary

Whenever opening up the artifacts UI tab, a preview is shown first, before allowing you to click the blue "download" button.

This can be troublesome for large csvs (dataframes). The browser would pull 50mb and render a preview, and then allow to download the file (re-downloading all the mb).

A "file size too large, do you want to load a preview anyway?" Dialogue could be shown instead, and allow just downloading the file without loading the preview.

### Motivation

-

### What component(s) does this bug affect?

- [X] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [X] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [X] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by dbczumar** [comment]
@ntakouris MLflow currently limits the amount of data that can be rendered in the preview to 50MB. I agree that 50MB might be too large. Happy to drop this to ~ 20MB. Would you be up for contributing a one-line PR to do this? https://github.com/mlflow/mlflow/blob/master/mlflow/server/js/src/experiment-tracking/components/artifact-view-components/ShowArtifactPage.js#L29

**Comment by ntakouris** [comment]
Yeah @dbczumar , but how do we decide how big is too big? I'd say anything more the size of a FHD jpeg would be large (2?3 mb?)
